network/gateway/complete.go

```diff
@@ -138,20 +141,22 @@ func (g *Complete) requestCertSign(ctx context.Context, discoveryNode insolar.Di
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		msg := &message.NodeSignPayload{
-			NodeRef: registeredNodeRef,
-		}
-		opts := &insolar.MessageSendOptions{
-			Receiver: discoveryNode.GetNodeRef(),
-		}
-		r, err := g.MessageBus.Send(ctx, msg, opts)
-		if err != nil {
-			return nil, err
-		}
-		sign = r.(reply.NodeSignInt).GetSign()
+		return sign, nil
 	}
 
+	request := &packet.SignCertRequest{
+		NodeRef: *registeredNodeRef,
+	}
+	future, err := g.HostNetwork.SendRequest(ctx, types.SignCert, request, *discoveryNode.GetNodeRef())
+	p, err := future.WaitResponse(10 * time.Second)
+	if err != nil {
+		return nil, err
+	} else if p.GetResponse().GetError() != nil {
+		return nil, fmt.Errorf("[requestCertSign] Remote said %s", p.GetResponse().GetError().Error)
```

Comment:
I think we should add reference of remote node that returned error to the log message

Reply:
ok